app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                    'cback_url' => action('Webapp\\PostController@getEdit', $post->id),
+                    'old_post_id' => $oldPostId,
+                    'priority' => $post->priority,
+                ]))->send($action)->getResult();
+                if ($returnData->data->post_id == 0) {
+                    $this->activityLog->setInfo('Post to target site failed for unknown reason');
+                    $this->activityLog->addMeta('FailedPostToTargetSite','unknown reason');
+                    throw new \Exception();
+                }
+                $post->content = Post::replaceIDButtonLink($returnData->data->post_id, $postContent, $oldPostId);
+                $post->target_post_id = $returnData->data->post_id;
+                $post->published_status = ($submitType == 'publish' && $post->site_id > 0) ? 1 : 0;
+            }
+
+            $post->site_id = $newSiteId;
+            $post->category = $newCategory;
+            $this->_flashMessage($request, $submitType == 'publish' ? '公開しました' : '下書きに保存しました', 'success');
+            $this->activityLog->addMeta('SiteId', $post->site_id);
+            $this->activityLog->addMeta('TargetPostId', $post->target_post_id);
+            $this->activityLog->addMeta('Category', $post->category);
+
+        } catch(BreakException $e) {
+            $this->_flashMessage($request, $submitType == 'publish' ? '公開しました' : '下書きに保存しました', 'success');
+        } catch(TitleDuplicatedException $e) {
+            $this->_flashMessage($request, $e->getMessage(), 'danger');
+            $this->activityLog->addMeta('Error', 'Title duplicated: '.$post->title);
+            $post->title = $oldTitle;
+            $post->site_id = $oldSiteId;
+        } catch(\Exception $e) {
+            $post->content = Post::replaceIDButtonLink('__POSTID__', $postContent, $oldPostId);
+            if($e->getMessage() == 'POST_NOT_FOUND') {
+                $post->target_post_id = 0;
+                $post->published_status = 0;
+            }
+            $this->_flashMessage($request, '更新に失敗しました。もう一度ご確認ください。' . $e->getMessage(), 'warning');
+            $this->activityLog->setInfo('Post to target site failed. Error: '.$e->getMessage());
+            $this->activityLog->addMeta('FailedPostToTargetSite', $e->getMessage());
+        }
+
+        $post->save();
+        $post->saveHistory();
+
+        $this->activityLog->addMeta('NewPublishedStatus', $post->published_status);
+        $this->activityLog->addMeta('NewSiteId', $post->site_id);
+
+        // Update Post meta and keywords
+        // Main keyword
+        PostMeta::deletePostMetaByPostId($post->id);
+        $main = explode(',', $request->get('mainkw'));
+
+        $uniqueMainKeyword = trim($main[0]);
+        $mainKeyword = Keywords::getKeyword($uniqueMainKeyword, 1);
+
+        if(empty($mainKeyword)) {//there is no keyword
+            if(strlen($uniqueMainKeyword) > 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1066)